### PR TITLE
[release-0.5] Update AKS test version to v1.22

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What type of PR is this?**
/kind failing-test

**What this PR does / why we need it**: Fix AKS tests that are failing as 1.19 was removed. This will buy us some time until we change it to using the "latest" stable release. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/config/azure-dev.yaml [hunk 1]
```diff
@@ -97,7 +97,7 @@ providers:
 
 variables:
   KUBERNETES_VERSION: "${KUBERNETES_VERSION:-v1.22.1}"
-  AKS_KUBERNETES_VERSION: "${KUBERNETES_VERSION:-v1.19.9}"
+  AKS_KUBERNETES_VERSION: "${KUBERNETES_VERSION:-v1.22.6}"
```
Comment (CecileRobertMichon) [inline test/e2e/config/azure-dev.yaml:100]: it says 1.22.6 but the code actually uses the latest 1.22 version (we changed it to the "stable-" pattern later in 1.0.

**Comment by CecileRobertMichon** [comment]
/retest

AKS cluster test passed

**Comment by CecileRobertMichon** [comment]
```
compute.VirtualMachinesClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: Code=\"InternalOperationError\" Message=\"An internal error occurred in the operation.\""
```

some VMs are failing to come up

**Comment by CecileRobertMichon** [comment]
AKS flake :(

/retest

**Review by mboersma — APPROVED** [review]
/lgtm

**Pull request merged**